pkgs/misc/fish-plugins/build-fish-plugin.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, lib, writeShellScriptBin, writeScript, fish }:
+
+let
+  rtpPath = "share/fish";
+
+  mapToFuncPath = v:
+    if lib.isString v
+    then v
+    else "${v}/${rtpPath}/vendor_functions.d";
+
+  fishWithFunctionPath = plugins: let
```

Comment:
What about using this function to define a `fish.withPlugins`? It would be useful for quickly testing plugins and running them inside a nix-shell.

Reply:
Good idea, this would be useful.

I think it can be done by creating some `wrapFish` package which would accept lists of plugins, custom functions, completion scripts, initialisation scripts and environment variables as customisation parameters.

This is a bit involved though and I'd prefer to keep that for a separate pull request.

Reply:
> This is a bit involved though and I'd prefer to keep that for a separate pull request.

Yeah, I agree.